app/components/cases/submissions/notifications.js

```diff
@@ -90,68 +104,76 @@ export default class CasesSubmissionsNotificationsComponent extends Component {
 
   @action
   addApprovalAddress(address, insertAsFirst) {
-    if (!this.notificationData.approvalAddresses.includes(address)) {
-      let newNotificationData = this.notificationData;
+    if (!this.approvalAddresses.includes(address)) {
       if (insertAsFirst === true) {
-        newNotificationData.approvalAddresses.splice(0, 0, address)
+        this.approvalAddresses.splice(0, 0, address)
       } else {
-        newNotificationData.approvalAddresses.push(address);
+        this.approvalAddresses.push(address);
+      }
+      if (!this.updateDefaultAddresses.isRunning) {
+        this.args.onNotificationDataChanged?.(this.notificationData);
```

Comment:
explanation regarding this:
When calculating the default we were calling the parent multiple times to set our new data.
But the getter `notificationData` was dependent on the data from args.
So we were updating the `notificationData` we are using to calculate the `notificationData`.
on Init, only call the parent once after everything is calculated, then we can update individual parts.